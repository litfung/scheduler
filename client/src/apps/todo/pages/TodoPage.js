import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { createBrowserHistory } from 'history';
// import ReactSVG from 'react-svg';
import { Redirect } from 'react-router-dom';

let __isNode__ = false;
if (typeof process === 'object') {
	if (typeof process.versions === 'object') {
		if (typeof process.versions.node !== 'undefined') {
			__isNode__ = true;
		}
	}
}

const TodoPage = (props) => {
	// Getting 'req' from staticContext

	// const { staticContext, groups, events } = props;
	// const req = __isNode__ && staticContext ? staticContext.req : undefined;

	// // To Find Query Str in the client
	// const urlParams = !__isNode__ ? new URLSearchParams(window.location.search) : undefined;

	// // Returns Page Initial Group-Id
	// const initGroupId = (req, groups) => {
	// 	const hasGroups = groups && groups.length > 0;
	// 	if (__isNode__ && req) {
	// 		return req.query.group || hasGroups ? groups[0]._id.toString() : '';
	// 	} else {
	// 		return urlParams.get('group') || hasGroups ? groups[0]._id.toString() : '';
	// 	}
	// };

	// // Returns Page Initial Event-Id
	// const initEventId = (req) => {
	// 	if (__isNode__ && req) {
	// 		return req.query.event || '';
	// 	} else {
	// 		return urlParams.get('event') || '';
	// 	}
	// };

	// Component States
	// const [ groupId, setGroupId ] = useState(initGroupId(req, groups));
	// const [ eventId, setEventId ] = useState(initEventId(req));

	// // Lifecycle
	// useEffect(() => {
	// 	if (!__isNode__) {
	// 		setGroupId(urlParams.get('group') || '');
	// 		setEventId(urlParams.get('event') || '');

	// 		if (!urlParams.get('group') && props.groups[0]) {
	// 			setGroupId(props.groups[0]._id);
	// 		}
	// 	}

	// 	return () => {};
	// }, []);

	// const changeGroupId = (id) => {
	// 	setGroupId(id);
	// 	setEventId('');
	// 	const history = createBrowserHistory();
	// 	history.push(`/todo?group=${id}`);
	// };

	// const changeEventId = (id) => {
	// 	// setGroupId('');
	// 	setEventId(id);
	// 	const history = createBrowserHistory();
	// 	history.push(`/todo?group=${groupId}&event=${id}`);
	// };

	// const activeGroup = props.groups.find(({ _id }) => _id === groupId);
	// const activeEvent = props.events.find(({ _id }) => _id === eventId);

	return <div>Todo</div>;

	return (
		<div className="todo-page-000">
			<TodoSidebarComp changeGroupId={changeGroupId} active_groupId={groupId} visible={props.sideBar} />
			<div className="todo-page-001-content">
				{!groupId ? (
					// <h2><ReactSVG src='/logo.svg'/>Loading...</h2>
					<h2>Loading...</h2>
				) : (
					<TodoListComp active_groupId={groupId} activeGroup={activeGroup} changeEventId={changeEventId} />
				)}
			</div>
			{activeEvent && (
				<EventDetailsComp
					event={activeEvent}
					hex_color={activeGroup.hex_color}
					closeEventDetails={() => {
						setEventId('');
						const history = createBrowserHistory();
						history.push(`/todo?group=${groupId}`);
					}}
				/>
			)}
		</div>
	);
};

export default {
	component: TodoPage,
	loadData: function(store) {
		// return store.dispatch(asyncReadAllGroups());
	}
};
