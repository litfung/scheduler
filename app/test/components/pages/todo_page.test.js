import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';

import events from '../../__data__/events';
import groups from '../../__data__/groups';

import TodoPage from '../../../src/client/components/pages/todo_page';
import configureStore from '../../../src/helpers/configure_store';

test('Snapshot for TodoPage Component', () => {
  const wrapper = shallow(
    <Provider store={configureStore({ get: jest.fn(() => {}) })}>
      <TodoPage.component
        auth={{}}
        sideBar={true}
        handleAppMode={jest.fn(() =>  {})}
        setReduxCalendar={jest.fn(() =>  {})}
        groups={groups}
        events={events}
      />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
