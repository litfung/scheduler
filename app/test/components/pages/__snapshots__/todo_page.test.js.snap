// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for TodoPage Component 1`] = `
<Connect(TodoPage)
  auth={Object {}}
  events={
    Array [
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_group": "5c375e536f0de60df8c62f9e",
        "_id": "5c3e37f1b029023bcc7a4e81",
        "_isDeleted": false,
        "_isDone": false,
        "_rank": 15,
        "date_from": 1547490600000,
        "date_to": 1547490600000,
        "description": "",
        "hex_color": "#2ecc71",
        "notification": false,
        "title": "ndnv",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_group": "5c38b60d5a84ae04e85e9d21",
        "_id": "5c3e37f4b029023bcc7a4e83",
        "_isDeleted": false,
        "_isDone": false,
        "_rank": 16,
        "date_from": 1547490600000,
        "date_to": 1547490600000,
        "description": "",
        "hex_color": "#2ecc71",
        "notification": false,
        "title": "jdavjd",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_group": "5c373f59d64892352c703256",
        "_id": "5c3e9c5a2e22563e78433c4d",
        "_isDeleted": false,
        "_isDone": false,
        "_rank": 52,
        "date_from": 1547577000000,
        "date_to": 1547577000000,
        "description": "",
        "hex_color": "#59abe3",
        "notification": false,
        "title": "What",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_group": "5c373f59d64892352c703256",
        "_id": "5c3e9ebdbd71cf35cc3f746f",
        "_isDeleted": false,
        "_isDone": false,
        "_rank": 35,
        "date_from": 1539109800000,
        "date_to": 1540146600000,
        "description": "",
        "hex_color": "#59abe3",
        "notification": false,
        "title": "dvdds",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_group": "5c373f59d64892352c703256",
        "_id": "5c3e9ec4bd71cf35cc3f7471",
        "_isDeleted": false,
        "_isDone": false,
        "_rank": 50,
        "date_from": 1548009000000,
        "date_to": 1548009000000,
        "description": "",
        "hex_color": "#59abe3",
        "notification": false,
        "title": "Yh",
      },
    ]
  }
  groups={
    Array [
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c373f59d64892352c703256",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": true,
        "_rank": 0,
        "hex_color": "#4286f4",
        "title": "Tasks",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c3742106f0de60df8c62f82",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": false,
        "_rank": 3,
        "hex_color": "#f7ca18",
        "title": "Todos",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c38b60e5a84ae04e85e9d22",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": false,
        "_rank": 5,
        "hex_color": "#2ecc71",
        "title": "x",
      },
      Object {
        "__v": 0,
        "_creator": "5c373f59d64892352c703255",
        "_id": "5c3fb10855de832ac8e374b2",
        "_isDeleted": false,
        "_isOnCalendar": true,
        "_isPermanent": false,
        "_rank": 10,
        "hex_color": "#2ecc71",
        "title": "New",
      },
    ]
  }
  handleAppMode={[MockFunction]}
  setReduxCalendar={[MockFunction]}
  sideBar={true}
/>
`;
